Feature - 2790 - Add New Search Container/Form

esizer opened the pull request:
Resolves #2790 

## Summary

This duplicates the existing `SearchContainer` and `SearchForm` putting them behind the feature flag `FEATURE_APPLICANTSEARCH`.

## Testing

1. Rebuild your php container
2. Build the `talentsearch` frontend
3. Navigate to `/talent/search`
4. Test the form (it should function no differently)

petertgiles commented:
LOL, there's a bit more than just copying the component going on here. :laughing: Could you give me a guide to the refactoring that happened here to help me follow?

esizer commented:
> LOL, there's a bit more than just copying the component going on here. 😆 Could you give me a guide to the refactoring that happened here to help me follow?

Yes, git did some interesting things with +/- that I don't understand which makes it seem more complicated than it really is.

- Extracted [CandidateResults](https://github.com/GCTC-NTGC/gc-digital-talent/blob/8c75d14458e8633b1c06eaa2cb623bdd78090cf0/frontend/talentsearch/src/js/components/search/CandidateResults.tsx) into own file.
- Extracted [FilterBlock](https://github.com/GCTC-NTGC/gc-digital-talent/blob/8c75d14458e8633b1c06eaa2cb623bdd78090cf0/frontend/talentsearch/src/js/components/search/FilterBlock.tsx) into own file
- Updated function props to use a more conventional `on` prefix
- Removed `updateInitialValues` since it was not in use
- Updated to prefer default exports

petertgiles reviewed (approved):
Very nice!

esizer commented on frontend/talentsearch/src/js/components/search/SearchContainer.tsx:
```diff
@@ -203,62 +181,32 @@ export const SearchContainer: React.FC<SearchContainerProps> = ({
-export const SearchContainerApi: React.FC = () => {
-  const [initialValues, setInitialValues] = useState<FormValues | null>(null);
+const SearchContainerApi: React.FC = () => {
+  const [initialValues, setInitialValues] = React.useState<FormValues | null>(
```
removed with[remove unused search state](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3109/commits/560619be4195961b055706006e3c8ffc79d13e58)
